addons/website/models/website.py

```diff
@@ -1379,13 +1399,23 @@ def _remove_port(domain_name):
         def _filter_domain(website, domain_name, ignore_port=False):
             """Ignore `scheme` from the `domain`, just match the `netloc` which
             is host:port in the version of `url_parse` we use."""
-            website_domain = get_base_domain(website.domain)
+            website_domain = get_base_domain(website.domain_punycode)
             if ignore_port:
                 website_domain = _remove_port(website_domain)
                 domain_name = _remove_port(domain_name)
             return website_domain.lower() == (domain_name or '').lower()
 
-        found_websites = self.search([('domain', 'ilike', _remove_port(domain_name))])
+        # We need to test two possibilities idna or unicode (safety guard)
```

Comment:
"idna or unicode" seems misleading, for me it's the same thing. I'd say "punycode or unicode" or "punycode or idna" or "ascii or unicode", anything :wink: